### PR TITLE
Create Template for NIM deployment & Check KServe Installation before enabling NIM

### PR DESCRIPTION
**Pull request by xieshenzh**
## Description

Create a Template for NIM ServingRuntime when NIM app is enabled.
The Template should not be shown in the `Serving Runtime` list of `Settings`.

Check KServe installation in the status of DataScienceCluster before validating ngc api key, when enabling NIM app.

## How Has This Been Tested?

Negative scenario 1:
1. Enter an invalid ngc api key to enable NIM
2. Check if the Job for validating the api key fails
3. Check if the NIM application is not enabled
4. Check if the Template `nvidia-nim-serving-template` doesn't exist

Negative scenario 2:
1. Not enable KServe for Openshift AI installation.
2. Enter a valid ngc api key to enable NIM
3. Check if the Job for validating the api key fails
4. Check if the NIM application is not enabled

Positive scenario:
1. Enable KServe for Openshift AI installation.
2. Enter a valid ngc api key to enable NIM
3. Check if the Job for validating the api key succeeds
4. Check if the NIM application is enabled
5. Check if the Template `nvidia-nim-serving-template` is created
6. Check if Template `nvidia-nim-serving-template` is not shown in the `Serving Runtimes` list of `Settings

## Test Impact

No impact on existing code

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [x] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by xieshenzh** [comment]
This PR should be rebased and merged after https://github.com/opendatahub-io/odh-dashboard/pull/2959 is merged.

**Comment by mpaulgreen** [comment]
@xieshenzh @andrewballantyne can this PR be merged now that 2959 got merged?

File: manifests/rhoai/shared/base/nvidia-nim-validator-cron.yaml [hunk 1]
```diff
@@ -33,6 +33,105 @@ spec:
...
+                      },
+                      "name": "nvidia-nim-serving-template"
+                    },
+                    "objects": [
+                      {
+                        "apiVersion": "serving.kserve.io/v1alpha1",
+                        "kind": "ServingRuntime",
+                        "labels": {
+                          "opendatahub.io/dashboard": "true"
+                        },
```
Comment (mpaulgreen) [inline manifests/rhoai/shared/base/nvidia-nim-validator-cron.yaml:54]: @xieshenzh https://redhat-internal.slack.com/archives/C07CZDA6DE1/p1725994445465669?thread_ts=1725991550.539169&cid=C07CZDA6DE1

Comment (xieshenzh) [reply manifests/rhoai/shared/base/nvidia-nim-validator-cron.yaml:54]: This label is for ServingRuntime, not for the Template. I removed the label for the Template.
If my understanding is correct, Templates without the label should not be shown in `Settings` -> `Serving Runtimes` list. And I tested it in our lab last week, it worked as expected.

The NIM ServingRuntime are supposed to be created with this label via dashboard, right?

**Review by andrewballantyne — APPROVED** [review]
Looks like a great addition to keep things balanced. I know @xieshenzh is out on PTO, so lets get this in and we can look at improvements as time goes on

**Pull request merged**